torch/fx/annotate.py

```diff
@@ -0,0 +1,13 @@
+from torch.fx.proxy import Proxy
+
+def annotate(val, type):
+    # val could be either a regular value (not tracing)
+    # or fx.Proxy (tracing)
+    if isinstance(val, Proxy):
+        if val.node.type:
+            raise RuntimeError("type already exists")
```

Comment:
A good error message will direct the user on what they can do to fix the issue. I recommend something like

```
raise RuntimeError(f"Tried to annotate a value that already had a type on it! Existing type is {val.node.type} "
                   f"and new type is {type}. This could happen if you tried to annotate a function parameter "
                   f"value (in which case you should use the type slot on the function signature) or you called "
                   f"annotate on the same value twice")
```